packages/react-composites/src/composites/ChatComposite/file-sharing/FileUpload.ts

```diff
@@ -96,41 +100,30 @@ export interface FileUploadManager {
 }
 
 /**
- * An internal interface used by the Chat Composite to drive the UI for file uploads.
- * @beta
+ * A wrapper object for a file that is being uploaded.
+ * Provides common functions for updating the upload progress, canceling an upload etc.
+ * @internal
  */
-export interface ObservableFileUpload extends FileUploadEventEmitter {
-  /**
-   * Unique identifier for the file upload.
-   */
-  id: string;
+export class FileUpload implements FileUploadManager, FileUploadEventEmitter {
+  private _emitter: EventEmitter;
+  public id: string;
+  public file?: File;
   /**
    * Filename to be displayed in the UI during file upload.
    */
-  fileName: string;
+  public fileName: string;
   /**
    * Optional object of type {@link FileMetadata}
    */
-  metadata?: FileMetadata;
-}
-
-/**
- * A wrapper object for a file that is being uploaded.
- * Provides common functions for updating the upload progress, canceling an upload etc.
- * @internal
- */
-export class FileUpload implements FileUploadManager, ObservableFileUpload {
-  private _emitter: EventEmitter;
-  public id: string;
-  public file: File;
-  fileName: string;
+  public metadata?: FileMetadata;
 
-  constructor(file: File, maxListeners = _MAX_EVENT_LISTENERS) {
+  constructor(data: { file?: File; metadata?: FileMetadata }) {
```

Comment:
finding it hard to discern; will this ever be constructed with both a file and metadata?
If not can this instead be `constructor(uploadData: File | FileMetaData)` and the property be `public readonly fileUpload: File | FileMetaData` ?

Reply:
Or alternatively, should `notifyUploadCompleted` be mutating and setting the metadata field?

Reply:
Simplified the constructor which in turn simplified it's underlying usage.
Check it out :)